model/project_ref.go

```diff
@@ -1489,3 +1469,92 @@ func GetUpstreamProjectName(triggerID, triggerType string) (string, error) {
 	}
 	return upstreamProject.DisplayName, nil
 }
+
+func projectRefPipelineForValueIsBool(projectKey, repoKey string, val bool) []bson.M {
+	return []bson.M{
+		lookupRepoStep,
+		{"$match": bson.M{
+			"$or": []bson.M{
+				{projectKey: val},
+				{projectKey: nil, bsonutil.GetDottedKeyName("repo_ref", repoKey): val},
+			},
+		}},
+	}
+}
+
+func projectRefPipelineForMatchingTrigger(project string) []bson.M {
+	return []bson.M{
+		lookupRepoStep,
+		{"$match": bson.M{
+			"$and": []bson.M{
+				{"$or": []bson.M{
+					{ProjectRefEnabledKey: true},
+					{ProjectRefEnabledKey: bson.M{"$ne": false}, bsonutil.GetDottedKeyName("repo_ref", RepoRefEnabledKey): true},
+				}},
+				{"$or": []bson.M{
+					{
+						bsonutil.GetDottedKeyName(projectRefTriggersKey, triggerDefinitionProjectKey): project,
+					},
+					{
+						projectRefTriggersKey: nil,
+						bsonutil.GetDottedKeyName("repo_ref", RepoRefTriggersKey, triggerDefinitionProjectKey): project,
+					},
+				}},
+			}},
+		},
+	}
+}
+
+func projectRefPipelineForCommitQueueEnabled() []bson.M {
+	return []bson.M{
+		lookupRepoStep,
+		{"$match": bson.M{
+			"$and": []bson.M{
+				{"$or": []bson.M{
+					{ProjectRefEnabledKey: true},
+					{ProjectRefEnabledKey: nil, bsonutil.GetDottedKeyName("repo_ref", RepoRefEnabledKey): true},
+				}},
+				{"$or": []bson.M{
+					{
+						bsonutil.GetDottedKeyName(projectRefCommitQueueKey, commitQueueEnabledKey): true,
+					},
+					{
+						bsonutil.GetDottedKeyName(projectRefCommitQueueKey, commitQueueEnabledKey):          nil,
+						bsonutil.GetDottedKeyName("repo_ref", RepoRefCommitQueueKey, commitQueueEnabledKey): true,
+					},
+				}},
+			}},
+		},
+	}
+}
+
+func projectRefPipelineForPeriodicBuilds() []bson.M {
```

Comment:
could you add a short comment explaining what each of these functions do? I think they're complex enough that you can't fit it all in the name, and it'll specifically help this function since its logic is different than the ones above

Reply:
Good call